apps/web/app/(app)/environments/[environmentId]/surveys/[surveyId]/(analysis)/summary/components/shareEmbedTabs/EmailTab.tsx

```diff
@@ -0,0 +1,396 @@
+import { sendEmailAction } from "@/app/(app)/environments/[environmentId]/surveys/[surveyId]/(analysis)/summary/actions";
+import CodeBlock from "@/components/shared/CodeBlock";
+import { cn } from "@formbricks/lib/cn";
+import { QuestionType } from "@formbricks/types/questions";
+import { TProduct } from "@formbricks/types/v1/product";
+import { TProfile } from "@formbricks/types/v1/profile";
+import { TSurvey } from "@formbricks/types/v1/surveys";
+import { Button, Input } from "@formbricks/ui";
+import { CodeBracketIcon, DocumentDuplicateIcon, EnvelopeIcon } from "@heroicons/react/24/solid";
+import {
+  Column,
+  Container,
+  Button as EmailButton,
+  Link,
+  Row,
+  Section,
+  Tailwind,
+  Text,
+  render,
+} from "@react-email/components";
+import { useMemo, useState } from "react";
+import toast from "react-hot-toast";
+
+interface EmailTabProps {
+  survey: TSurvey;
+  surveyUrl: string;
+  profile: TProfile;
+  product: TProduct;
+}
+
+export default function EmailTab({ survey, surveyUrl, profile, product }: EmailTabProps) {
+  const [email] = useState(profile.email);
+  const [showEmbed, setShowEmbed] = useState(false);
+  const brandColor = product.brandColor;
+  const subject = "Formbricks Email Survey Preview";
+
+  const PreviewEmailTemplate = useMemo(() => {
+    return getEmailTemplate(survey, surveyUrl, brandColor, true);
+  }, [survey, surveyUrl]);
+
+  const EmailTemplate = useMemo(() => {
+    return getEmailTemplate(survey, surveyUrl, brandColor, false);
+  }, [survey, surveyUrl]);
+
+  const previewEmailHtml = render(PreviewEmailTemplate, { pretty: true });
+  const emailHtml = render(EmailTemplate, { pretty: true });
+
+  const previewEmailHtmlWithoutDoctype = useMemo(
+    () =>
+      previewEmailHtml.replace(
+        '<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">',
+        ""
+      ),
+    [previewEmailHtml]
+  );
+  const emailHtmlWithoutDoctype = useMemo(
+    () =>
+      emailHtml.replace(
+        '<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">',
+        ""
+      ),
+    [emailHtml]
+  );
+
+  const sendPreviewEmail = async () => {
+    await sendEmailAction({ html: previewEmailHtmlWithoutDoctype, subject, to: email });
```

Comment:
there's no error handling here, while this action is throwing an error, please wrap this in a try catch like this:
```js
const sendPreviewEmail = async () => {
    try {
      await sendEmailAction({ html: previewEmailHtmlWithoutDoctype, subject, to: email });
      toast.success("Email sent!");
    } catch (err) {
      if (err instanceof AuthenticationError) {
        toast.error("You are not authenticated to perform this action.");
        return;
      }

      toast.error("Something went wrong. Please try again later.");
    }
  };
  ```